dataservice/api/biospecimen/resources.py

```diff
@@ -56,6 +58,10 @@ def get(self, filter_params, after, limit):
         if diagnosis_id:
             q = (q.join(BiospecimenDiagnosis)
                  .filter(BiospecimenDiagnosis.diagnosis_id == diagnosis_id))
+        if genomic_file_id:
+            q = (q.join(BiospecimenGenomicFile)
+                 .filter(BiospecimenGenomicFile.genomic_file_id ==
+                 genomic_file_id))
```

Comment:
Any chance of this causing duplication of the genomic files when there are multiple biospecimens that relate to it?

Reply:
I don't think so... When you do the join between Biospecimen and BiospecimenGenomicFile the result can for sure have duplicate biospecimens or genomic files bc of the many-to-many. But because we only do this join with a filter by genomic_file_id, we will never have duplicate biospecimens in the result since we have the unique constraint on (biospecimen_id, genomic_file_id). The same is true for genomic_file/resources.py.